daemon/snapper/timestamp_formatting/timestamp_formatting.go

```diff
@@ -0,0 +1,92 @@
+package timestamp_formatting
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type Formatter struct {
+	format   func(time.Time) string
+	location *time.Location
+}
+
+func New(formatString string, locationString string) (*Formatter, error) {
+	location, err := time.LoadLocation(locationString) // no shadow
+	if err != nil {
+		return nil, errors.Wrapf(err, "load location from string %q", locationString)
+	}
+	makeFormatFunc := func(formatString string) (func(time.Time) string, error) {
+		if strings.Contains(formatString, "+") {
+			return nil, fmt.Errorf("character '+' is not allowed in ZFS snapshot names")
+		}
+		return func(t time.Time) string {
+			res := t.Format(formatString)
+			// if the formatString contains a time zone specifier
+			// and the location would result in a positive offset to UTC
+			// then the result of t.Format would contain a '+' sign.
+			if isLocationPositiveOffsetToUTC(location) {
```

Comment:
If format against "all sorts of things" (check prev comment) is not guarded, resulting snapshot name can contain prohibited characters, so the only source of errors is not really an offset.

I would still use regexp as I used in my PR, maybe print a warning that there were invalid characters.

Reply:
>  so the only source of errors is not really an offset.

Addressed with recent pushes ; we now do the same check that ZFS does.

Reply:
> I would still use regexp as I used in my PR, maybe print a warning that there were invalid characters.

We now detect this shortly after config parsing during startup, and will fail to start the daemon.